state/indexer/sink/psql/psql_test.go

```diff
@@ -3,195 +3,250 @@ package psql
 import (
 	"context"
 	"database/sql"
-	"errors"
+	"flag"
 	"fmt"
 	"io/ioutil"
+	"log"
 	"os"
+	"os/signal"
 	"testing"
 	"time"
 
-	sq "github.com/Masterminds/squirrel"
-	schema "github.com/adlio/schema"
+	"github.com/adlio/schema"
 	proto "github.com/gogo/protobuf/proto"
-	_ "github.com/lib/pq"
-	dockertest "github.com/ory/dockertest"
+	"github.com/ory/dockertest"
 	"github.com/ory/dockertest/docker"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	abci "github.com/tendermint/tendermint/abci/types"
 	"github.com/tendermint/tendermint/state/indexer"
 	"github.com/tendermint/tendermint/types"
+
+	// Register the Postgres database driver.
+	_ "github.com/lib/pq"
 )
 
-var db *sql.DB
-var resource *dockertest.Resource
-var chainID = "test-chainID"
+// Verify that the type satisfies the EventSink interface.
+var _ indexer.EventSink = (*EventSink)(nil)
 
 var (
+	doPauseAtExit = flag.Bool("pause-at-exit", false,
+		"If true, pause the test until interrupted at shutdown, to allow debugging")
+
+	db *sql.DB
```

Comment:
is there anyway to avoid having this be global? or is this something for the future? 

I dug up the test fixture stuff that I used/wrote before just for giggles, mostly https://github.com/tychoish/amboy/blob/main/queue/pgq/queue_test.go#L26-L75.

Reply:
There is, but I'm not sure it's necessarily worthwhile. We could make it somewhat less-magical by (1) giving this variable a better name (e.g. `testDB`) and/or giving ownership of it to a helper function (e.g., `func testDB() *sql.DB { … }` or similar). Neither of those gets rid of the global-ness, but both of those would make it easier for the reader to see what's going on.

That said: This is a test. IMO it's fine for shared resources across the tests to be global. That's arguably what globals are for.

Reply:
5970fb3c attempts to clarify the usage a bit, though it is still a shared (global) value under the covers.

Reply:
That's fair. I actually tend to think that globals like this in tests lead to accidental shared state between tests and faulty test results due to test ordering/parallelism/etc. 

This is probably fine for now, and certainly this critique of this code isn't about this change.

Reply:
I agree, and the tests we have here now already cannot be parallelized, though if they were we'd be relying on the fact that a database handle is safe for concurrent use. That does make me realize, though, that I should restructure the test to make the order dependency more explicit. (We're relying on the implicit ordering of the code generator)

Reply:
e3513f57 makes this more explicit by executing the order-dependent tests as subtests.